src/config.c

```diff
@@ -484,6 +484,21 @@ void loadServerConfigFromString(char *config) {
                 err = "wrong number of arguments";
                 goto loaderr;
             }
+
+            if ((config->flags & MULTI_ARG_CONFIG) && argc == 2) {
+                /* For MULTI_ARG_CONFIGs, if we only have one argument, try to split it by spaces.
+                 * So that we can support something like --config "arg1 arg2 arg3". */
+                sds *new_argv;
+                int new_argc;
+                new_argv = sdssplitargs(argv[1], &new_argc);
+                if (!config->interface.set(config, new_argv, new_argc, &err)) {
+                    goto loaderr;
+                }
+                sdsfreesplitres(argv, argc);
+                sdsfreesplitres(new_argv, new_argc);
+                continue;
+            }
+
             /* Set config using all arguments that follows */
             if (!config->interface.set(config, &argv[1], argc-1, &err)) {
                 goto loaderr;
```

Comment:
I think you can refactor a bit so the `sdsfreesplitres(argv,argc);` at the end is reused in both cases.
```suggestion
            } else if ((config->flags & MULTI_ARG_CONFIG) && argc == 2) {
                /* For MULTI_ARG_CONFIGs, if we only have one argument, try to split it by spaces.
                 * So that we can support something like --config "arg1 arg2 arg3". */
                sds *new_argv;
                int new_argc;
                new_argv = sdssplitargs(argv[1], &new_argc);
                if (!config->interface.set(config, new_argv, new_argc, &err)) {
                    goto loaderr;
                }
                sdsfreesplitres(new_argv, new_argc);
            } else {
                /* Set config using all arguments that follows */
                if (!config->interface.set(config, &argv[1], argc-1, &err)) {
                    goto loaderr;
            }                    
```

Reply:
I wrote it like this at first, and then I wanted to save that indentation.
yes, this way is better, updated